RELEASE_NOTES.md

```diff
@@ -31,6 +31,11 @@ v1.0.0-alpha.x
   keys for a given trait.
   [#498](https://github.com/OpenAssetIO/OpenAssetIO/issues/498)
 
+- Migrated the `ManagerInterface.preflight` method to C++.
+  [#588](https://github.com/OpenAssetIO/OpenAssetIO/issues/588)
+
+- Migrated the `Manager.preflight` method to C++.
```

Comment:
[fix] Needs a breaking change for the rename of the callback. Can we split that into it's own commit as its not clear from the title/message that this has been done too.

Reply:
Split into (first) commit da4fc8f0